### PR TITLE
Make `monitor::Actor` adhere to ADR013

### LINKED ISSUE #1149: Monitor actor does not follow ADR013 for blockchain events
[ADR013](https://github.com/itchysats/itchysats/blob/master/docs/adr/013-all-actors-can-hydrate-cfd.md) states that a CFD should be loaded and hydrated wherever needed.

The monitor actor does not follow this as it sends blockchain events to the `maker_cfd` / `taker_cfd` actor. Instead, it should use the `command::Executor` to call the corresponding command (`lock_tx_confirmed` etc) on the corresponding CFD.

The difficulty with making this happen is that our current test framework relies on sending the events directly to the `maker_cfd` and `taker_cfd` actor. This will have to be changed to interact with the monitor mock instead.

### PR DESCRIPTION
**Pull request by thomaseizinger**
No rush in merging this.

### CODE REVIEW

File: daemon/tests/happy_path.rs [hunk 1]
```diff
@@ -51,6 +50,90 @@ macro_rules! wait_next_state {
...
+            .confirm_close_transaction($id)
+            .await;
+    };
+    (cet, $id:expr, $maker:expr, $taker:expr) => {
+        $maker.mocks.monitor().await.confirm_cet($id).await;
+        $taker.mocks.monitor().await.confirm_cet($id).await;
+    };
+}
+
+macro_rules! expiry {
```
Comment (klochowicz) [inline daemon/tests/happy_path.rs]: nit: `expire`? (to align with macro above? 

Both could also use some simple doc comments, ie. "trigger expiry event in the monitor actor" or something like that

**Review by klochowicz — APPROVED** [review]
great refactor, thank you! 👍

File: daemon/tests/harness/mocks/monitor.rs [hunk 2]
```diff
@@ -40,6 +42,78 @@ impl MonitorActor {
...
+impl MockMonitor {
+    pub fn set_executor(&mut self, executor: command::Executor) {
+        self.executor = Some(executor);
+    }
+
+    pub async fn confirm_lock_transaction(&mut self, id: OrderId) {
+        self.executor
+            .as_ref()
+            .expect("executor to be set during test setup")
+            .execute(id, |cfd| Ok(cfd.handle_lock_confirmed()))
```
Comment (da-kami) [inline daemon/tests/harness/mocks/monitor.rs]: I find it a bit weird that we have to mock all of these. I wonder if we could come up with a design where we can use production code to achieve this, i.e. arrange the production code in a way that we trigger the executor, but don't use the state management in the production monitor.

**Review by da-kami — APPROVED** [review]

File: daemon/tests/harness/mocks/monitor.rs (hunk 2 above)
Comment (thomaseizinger) [reply daemon/tests/harness/mocks/monitor.rs]: Very interesting idea. We would have to make parts of the original monitor actor swappable.

The internal logic is already split into the IO component, updating the state sending out events. We would have to plug the mock in before the last bit.

Interesting to explore but also, not that important I feel? Esp. keeping monitor::Event private is something I'd like to keep.

**Review by rishflab — APPROVED** [review]

**Review by da-kami — APPROVED** [review]

**Comment by thomaseizinger** [comment]
@da-kami Is this an actual re-approval where you went through the code again or just a drive-by one? :P

It is quite the code change so I'd appreciate if people have a closer look!

**Pull request merged**